src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/DSA/DSAXml.cs

```diff
@@ -2,20 +2,926 @@
+                ),
+                (
+                    "aQuzepFF4F1ue0fEV4mKrt1yUBydFuebGtdahyzwF6qQu/uQ8bO39cA8h+RuhyVm" +
+                    "VSb9NBV7JvWWofCZf1nz5l78YVpVLV51acX/xFk9WgKZEQ5xyX4SIaWgP+mmk1rt" +
+                    "2I7ws7L3nTqZ7XX3uHHm6vJoDZbVdKX0wTus47S0TeE="
+                ),
+                "wCZ4AHd55S42BoIhS9R/j69CvC0=");
+        }
+
+        [ConditionalTheory(typeof(DSAFactory), nameof(DSAFactory.SupportsFips186_3))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void TestWriteDeficientXParameters(bool includePrivateParameters)
+        {
+            TestWriteXml(
+                DSATestData.Dsa2048DeficientXParameters,
+                includePrivateParameters,
+                (
+                    "lNPks58XJz6PJ7MmkvfDTTVhi9J5VItHNpOcK6TnKRFsrqxgAelXBcJ9fPDDCLyn" +
+                    "ArtHEtS7RIoQeLYxFQ6rRVyRZ4phxRwtNx4Cu6xw6cLG2nV7V4IOyP0JbhBew2wH" +
+                    "ik/X3Yck0nXTAH+U8S/5YWcpPGZ7RncH8dyafAs0vE/EdbFdUSQKeJpTpWtk1pwe" +
+                    "ArfKOtNBs7b2yYbx4GW/5oDYfe5tlZHY445Xw3rCmDsnlL6v/ix7W2ykm5gSSHMy" +
+                    "XGHeb4IEEQGL6XI/4r2oMywTCIqjKghtFNbwAgEBP1FnhkPzKswAUl2yLwAg2S+c" +
+                    "L0CIuNNaHnZNzYtwwLPS6w=="
+                ),
+                "23CgOhWOnMudk9v3Z5bL68pFqHA+gqRYAViO5LaYWrM=",
+                (
+                    "PPDxRLcKu9RCYNksTgMq3wpZjmgyPiVK/4cQyejqm+GdDSr5OaoN7HbSB7bqzveC" +
+                    "TjZldTVjAcpfmF74/3r1UYuN8IhaasVw/i5cWVYXDnHydAGUAYyKCkp7D5z5av1+" +
+                    "JQJvqAuflya2xN/LxBBeuYaHyml/eXlAwTNbFEMR1H/yHk1cQ8AFhHhrwarkrYWK" +
+                    "wGM1HuRCNHC+URVShpTvzzEtnljU3dHAHig4M/TxSeX5vUVJMEQxthvg2tcXtTjF" +
+                    "zVL94ajmYZPonQnB4Hlo5vcH71YU6D5hEm9qXzk54HZzdFRL4yJcxPjzxQHVolJv" +
+                    "e7ZNZWp7vf5+cssW1x6KOA=="
+                ),
+                (
+                    "cHLO4Kgw8hUDAviwzw8HFHtsaxMs5k309uE7nofw8txeBXRBGbaVgJU1GndCqeRc" +
+                    "cuI+6L8AmMgv0tB4fyGXRwv7DLwhRirTiT3vfBoN80/VKVf/AcafdsVkwmjrzUPe" +
+                    "w3bfU4qIdK807QB7TkbQZgBoE3kxqlmjLodbKUKdtVY13rbcjL+GfUSUytXt7n5/" +
+                    "IF7o6LLIoFK0Uo9HySsfjP7J7QU8IeOnMb/yaa0JnEE9X8h4U1EWXnmqehQ6DH5V" +
+                    "Ye8DvOPPDe2c7YMWgC+Z3a0DLejBknDzuvWoJgiIkX/Nc+Sx1W+tFWPHHbyS9nJW" +
+                    "kt3Wo5FBhE0R/aIwt75rrA=="
+                ),
+                // The rules from xmldsig say that these types are ds:CryptoBinary, which
+                // means they should leave off any leading 0x00 bytes.
+                //
+                // .NET Framework just treated it like base64Binary, though, and happily
+                // writes the unwanted zeroes.
+                "AMhxt+OJaF25fZNN1wEfqwdv8n7EKC+wDA1kbSnV5OU=");
+        }
+
+        [ConditionalFact(typeof(DSAFactory), nameof(DSAFactory.SupportsKeyGeneration))]
+        [OuterLoop("DSA key generation is very slow")]
```

Comment:
is it actually that slow that it's worth to put it in the outerloop?

Reply:
the odds of it failing are low, and it the delay it had was measured in seconds.

Reply:
fair enough